tardis/model.py

```diff
@@ -117,17 +118,23 @@ def __init__(self, tardis_config):
             v_inner=tardis_config.structure.v_inner,
             v_outer=tardis_config.structure.v_outer)
 
-        distance = tardis_config.supernova.get('distance', None)
-        self.spectrum = TARDISSpectrum(
-            tardis_config.spectrum.frequency, distance)
-        self.spectrum_virtual = TARDISSpectrum(
-            tardis_config.spectrum.frequency, distance)
-        self.spectrum_reabsorbed = TARDISSpectrum(
-            tardis_config.spectrum.frequency, distance)
-
         self.calculate_j_blues(init_detailed_j_blues=True)
         self.update_plasmas(initialize_nlte=True)
 
+    @property
+    @deprecated('v1.5', 'spectrum will be removed from model. Use model.runner.spectrum instead.')
+    def spectrum(self):
+        return self.runner.spectrum
+
+    @property
+    @deprecated('v1.5', 'spectrum_virtual will be removed from model. Use model.runner.spectrum_virtual instead.')
+    def spectrum_virtual(self):
+        return self.runner.spectrum_virtual
+
+    @property
```

Comment:
In the relevant PR (#586), yes, sure.

2016-06-13 18:03 GMT+03:00 Wolfgang Kerzendorf <notifications@github.com>:

> In tardis/model.py
> <https://github.com/tardis-sn/tardis/pull/587#discussion_r66807714>:
>
> >          self.calculate_j_blues(init_detailed_j_blues=True)
> >          self.update_plasmas(initialize_nlte=True)
> >
> > +    @property
> > +    @deprecated('v1.5', 'spectrum will be removed from model. Use model.runner.spectrum instead.')
> > +    def spectrum(self):
> > +        return self.runner.spectrum
> > +
> > +    @property
> > +    @deprecated('v1.5', 'spectrum_virtual will be removed from model. Use model.runner.spectrum_virtual instead.')
> > +    def spectrum_virtual(self):
> > +        return self.runner.spectrum_virtual
> > +
> > +    @property
>
> @ftsamis <https://github.com/ftsamis> can you attach a note to plasma as
> well.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/tardis-sn/tardis/pull/587/files/f924282c94957909eb57b33bc487b394bc92ad97#r66807714>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAFXPp2YmnI-1zn80xo89J2ZeHU7jnTWks5qLXEogaJpZM4IybYm>
> .
>